src/auxinfo/auxinfo_commit.rs

```diff
@@ -183,21 +188,21 @@ impl CommitmentScheme {
                 "Decommitment has the wrong session ID. Got {}, expected {}.",
                 self.sid, sid
             );
-            return Err(InternalError::ProtocolError);
+            return Err(InternalError::ProtocolError(Some(self.pid)));
         }
         if *sender != self.pid {
             error!(
                 "Decommitment has the wrong sender ID. Got {}, expected {}.",
                 self.pid, sender
             );
-            return Err(InternalError::ProtocolError);
+            return Err(InternalError::ProtocolError(Some(self.pid)));
         }
 
         let rebuilt_com = self.commit()?;
 
         if rebuilt_com != *com {
             error!("Commitment verification failed; does not match commitment. Commitment scheme: {:?}. Commitment: {:?}", self, com);
-            return Err(InternalError::ProtocolError);
+            return Err(InternalError::ProtocolError(Some(self.pid)));
```

Comment:
We would never blame ourselves for a problem; an inherent assumption of this implementation is that it's correct and is not cheating.

This method is verifying the `Commitment` received from another party. If it's invalid, it's the sender's fault, not ours.